power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/iterative_linear_se_solver.hpp

```diff
@@ -410,23 +414,31 @@ template <bool sym> class MeasuredValues {
     template <bool only_magnitude = false>
     static SensorCalcParam<sym> combine_measurements(std::vector<SensorCalcParam<sym>> const& data, Idx begin,
                                                      Idx end) {
-        double accumulated_inverse_variance = 0.0;
+        double accumulated_inverse_variance{};
         ComplexValue<sym> accumulated_value{};
         for (Idx pos = begin; pos != end; ++pos) {
-            accumulated_inverse_variance += 1.0 / data[pos].variance;
+            auto const& measurement = data[pos];
+            auto const inv_variance = std::isfinite(measurement.variance) ? 1.0 / measurement.variance : 0.0;
```

Comment:
part 1 of the fix

Reply:
Do we need this? `1.0/inf` is well defined in IEEE 754.

Reply:
yes because it returns `std::nan`, not `0.0`

Reply:
Are you sure? As far as I know, if you have `x / inf`, when `x` is not `inf` or `nan`, the result is garanteed to be `0.0`.

Reply:
just try to compile and run the following in different compilers

```c++
// Example program
#include <iostream>
#include <limits>


int main()
{
    std::cout << 1.0 / std::numeric_limits<double>::infinity();
}
```

Reply:
@mgovers is this merged without resolve this comment?

Reply:
sorry i forgot to comment (i did resolve the comment though; race condition???)

Nitish and I attempted it yesterday and it appeared to work as you said. Nitish applied the change in https://github.com/PowerGridModel/power-grid-model/pull/360/commits/90a16856e0d9a12a75cd8c3a59e6a5224c969597